sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -2634,6 +2644,82 @@ SIREPO.app.directive('bootstrapToggle', function() {
     };
 });
 
+
+SIREPO.app.directive('jobsList', function(requestSender, appState, $location, $sce) {
+    return {
+        restrict: 'A',
+        template: [
+            '<div>',
+                '<table class="table">',
+                '<thead>',
+                    '<th data-ng-repeat="c in data.header">{{ c }}</th>',
+                '</thead>',
+                '<tbody ng-bind-html="getRows()"></tbody>',
+                '</table>',
+                '<button class="btn btn-default" data-ng-click="getJobs()">Refresh</button>',
+            '</div>',
+        ].join(''),
+        controller: function($scope) {
+            function dataLoaded(data, status) {
+                $scope.data = data;
+            };
+
+            function getUrl(simulationId, app) {
+                return requestSender.formatUrlLocal(
+                    'source',
+                    {':simulationId': simulationId},
+                    app
+                );
+            };
+
+            function is_adm() {
+                return $location.path().includes('adm');
+            };
+
+            $scope.getJobs = function () {
+                requestSender.sendRequest(
+                    'jobs',
+                    dataLoaded,
+                    {
+                        adm: is_adm(),
+                        simulationType: SIREPO.APP_SCHEMA.simulationType,
+                    });
+            };
+
+            $scope.getRows = function() {
+                var d = $scope.data;
+                if (d) {
+                    var s = d.header.indexOf('Simulation id');
+                    var a = d.header.indexOf('App')
+                    if (s < 0 || a < 0) {
+                        throw new Error("'Simulation id' or 'App' not found on header=" + d.header);
+                    }
+                    var n = d.header.indexOf('Name')
+                    var m = is_adm();
+                    var o = '';
+                    for (var i in d.rows) {
+                        var r = d.rows[i];
+                        o += '<tr>';
+                        for (var j =0; j < r.length; j++ ) {
+                            var v = r[j];
```

Comment:
I think we hit our single letter variable limit. It would be good to add a function or some type of denormalization.